nautobot_circuit_maintenance/__init__.py

```diff
@@ -66,7 +71,7 @@ class CircuitMaintenanceConfig(PluginConfig):
 
     def ready(self):
         super().ready()
-        post_migrate.connect(custom_field_extension, sender=self)
+        post_migrate.connect(custom_fields_extension, sender=self)
```

Comment:
We really need a better pattern for the CF creation - as I don't think this signal will get re-triggered when upgrading to a new version of the plugin in isolation, since we don't have any new migrations to apply? I wonder if we should explicitly call `custom_fields_extension` from the circuits job as well so as to ensure that it's always up-to-date whenever the job gets run?

Ref: https://github.com/nautobot/nautobot/issues/13

Reply:
I see your point... but if I understand it correctly, calling it from the CM Job it's late as it will try to use the field, and only after it would be available, so it mitigates the issue but doesn't solves it.

I've reviewed the available signals from [here](https://docs.djangoproject.com/en/3.2/ref/signals/), and I don't see any better one than `post_migrate`. For the long term, to solve the issue that you referenced, maybe Nautobot could implement some custom signal when installing that this could be connected to.

For the short term (this PR), would be a solution to create a fake migration (without changes), just to make sure that this is triggered, and we could adopt the new pattern when proposed in Nautobot 1.2. WDYT?

Reply:
I'm probably missing some nuance - why would it be too late to call it as a first step in `HandleCircuitMaintenanceNotifications.run()`?

Reply:
Because the Job is using the content of the custom fields to customize the retrieval of notifications, so the it expects the value from these fields to be in place. 
My understanding is that the first time that the Job would be run, the Custom Fields would be created (if not done before from the post_migration signal), so then you could go to the `Provider` to update them, but the first execution of the Job would be useless

Reply:
Gotcha. The first execution of the job would follow the existing behavior (no flexible mapping) but after that first run it would be possible to add flexible mappings - which I think is still a decent fallback option, unless you can easily set up the necessary mock migration instead?

Reply:
trying 🥷 mode

Reply:
bf8befc worked like a charm (not very proud of me)
I've tested the migration path and works fine